tests/testsuite/progress.rs

```diff
@@ -53,28 +53,19 @@ fn bad_progress_config_missing_width() {
 }
 
 #[cargo_test]
-fn bad_progress_config_missing_when() {
+fn default_progress_is_auto() {
```

Comment:
Can you split the behavior change behind this test change into its own commit?  This change in behavior should be more explicitly acknowledged rather than conjoined in the same commit with the new feature that wants to leverage it

Reply:
Done: https://github.com/rust-lang/cargo/pull/15287